x-pack/platform/plugins/shared/lens/public/settings_storage.tsx

```diff
@@ -5,19 +5,67 @@
  * 2.0.
  */
 
-import type { IStorageWrapper } from '@kbn/kibana-utils-plugin/public';
+import { type IStorageWrapper, Storage } from '@kbn/kibana-utils-plugin/public';
 
 export interface LocalStorageLens {
   indexPatternId?: string;
   skipDeleteModal?: boolean;
+  userChartType?: string;
 }
 
 export const LOCAL_STORAGE_LENS_KEY = 'lens-settings';
 
+export const LENS_STORAGE_KEYS = {
+  USER_CHART_TYPE: 'userChartType',
+  INDEX_PATTERN_ID: 'indexPatternId',
+  SKIP_DELETE_MODAL: 'skipDeleteModal',
+} as const;
+
+/**
+ * Read a value from the Lens settings in localStorage
+ */
 export const readFromStorage = (storage: IStorageWrapper, key: string) => {
   const data = storage.get(LOCAL_STORAGE_LENS_KEY);
   return data && data[key];
 };
+
+/**
+ * Write a value to the Lens settings in localStorage
+ */
 export const writeToStorage = (storage: IStorageWrapper, key: string, value: string) => {
   storage.set(LOCAL_STORAGE_LENS_KEY, { ...storage.get(LOCAL_STORAGE_LENS_KEY), [key]: value });
 };
+
+/**
+ * Delete a value from the Lens settings in localStorage
+ */
+export const deleteFromStorage = (storage: IStorageWrapper, key: string) => {
+  const currentData = storage.get(LOCAL_STORAGE_LENS_KEY) || {};
+  const { [key]: deleted, ...rest } = currentData;
+  storage.set(LOCAL_STORAGE_LENS_KEY, rest);
+};
+
+// Convenience functions
+/**
+ * Get the user's preferred chart type from localStorage
+ */
+export const getUserChartTypeFromLocalStorage = (): string | null => {
+  const storage = new Storage(localStorage);
```

Comment:
I think it is better to use the `sessionStorage` instead of the `localStorage` to avoid conflicts and changes of the same variable across different tabs.
I tested and clicking Close on a tab besically removes the currently set viz type from the other tab and viceversa.

Reply:
Marco changed it to use session storage instead https://github.com/elastic/kibana/pull/231767/commits/211cf65764ba50d3416bd123e0b0d1cd7142419e